static/app/views/discover/table/index.tsx

```diff
@@ -204,6 +204,18 @@ class Table extends PureComponent<TableProps, TableState> {
       ? parseLinkHeader(pageLinks).previous.results === false
       : false;
 
+    let cursorVal: string | undefined = undefined;
+
+    const onCursorInternal = (
+      cursor: string | undefined,
+      path: string,
+      query: Query,
+      delta: number
+    ): void => {
+      cursorVal = cursor;
+      return onCursor(cursor, path, query, delta);
+    };
```

Comment:
I'm not a big fan of having this handler in the render method and changing a variable. I would probably prefer if we would reuse the `isFirstPage` boolean that's above this code block. Not a blocker, your call.

Reply:
Yes, re-working the code since the cursor is only set when there is a navigation done by the current user( i.e. the user presses prev/next buttons on the paginator), if the user opens a link at page 2 the cursor will not be set and the component will calculate the numSamples wrongly.
Looking how we get the cursor from the url since that is the proper place to look for it.